contracts/FriendsOnChainV2.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.9;
+
+// ██      ███████ ████████ ███████     ██████  ███████     ███████ ██████  ██ ███████ ███    ██ ██████  ███████
+// ██      ██         ██    ██          ██   ██ ██          ██      ██   ██ ██ ██      ████   ██ ██   ██ ██
+// ██      █████      ██    ███████     ██████  █████       █████   ██████  ██ █████   ██ ██  ██ ██   ██ ███████
+// ██      ██         ██         ██     ██   ██ ██          ██      ██   ██ ██ ██      ██  ██ ██ ██   ██      ██
+// ███████ ███████    ██    ███████     ██████  ███████     ██      ██   ██ ██ ███████ ██   ████ ██████  ███████
+
+// Import OpenZeppelin Contracts
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/ERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+
+// @title Friends on ChainV2
+// @author Ian Hunter (@ianh), Derek Brown (@derekbrown)
```

Comment:
Add your name here.

Reply:
We may also want to keep a public changelog somewhere in-code for upgradeable contracts?
Inline comments on why we moved to v2, for example.

Reply:
Rough stab:
```
@notice V1: Base Contract
@notice V2: Adds ability to add friends to a group post-mint.
```